sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImpl.java

```diff
@@ -350,19 +353,21 @@ public TokenRange(
    * Writer storing an entity into Apache Cassandra database.
    */
   protected class WriterImpl<T> implements Writer<T> {
-
+    private static final int CONCURRENT_ASYNC_QUERIES = 100;
```

Comment:
Does this number can be dependent on any configure option (like number of (v)nodes, connections, etc)?

Reply:
Not really, because it doesn't relate to the number of nodes or vnodes, it's more of a best practice to prevent nodes from being overwhelmed. The capacity of the nodes to handle a lot of concurrent queries will depend partly on the number of threads in the read thread pool of Cassandra, which is a configuration element we cannot access from the client.
I'm not sure users should be dealing with this as concurrency can already be handled by limiting the number of splits/workers.
We can make this configurable but it could be confusing to users and wouldn't bring improvements in throughput IMHO.

I'd be in favor of leaving this as a constant, but if you want to make it configurable I'll do it.

Reply:
Ok, let's keep it as it for now to avoid adding new tuning knobs (as it's not recommended in Beam - https://beam.apache.org/contribute/ptransform-style-guide/#what-parameters-to-expose)

Reply:
Though, it would make sense to add a comment why this number was chosen.

Reply:
+1 to constant + comment